src/Datadog.Trace/DuckTyping/DuckType.Statics.cs

```diff
@@ -32,7 +32,7 @@ public static partial class DuckType
         private static readonly MethodInfo _methodBuilderGetToken = typeof(MethodBuilder).GetMethod("GetToken", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
 
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
-        private static readonly Dictionary<Assembly, ModuleBuilder> ActiveBuilders = new Dictionary<Assembly, ModuleBuilder>();
+        private static readonly ConditionalWeakTable<Assembly, ModuleBuilder> ActiveBuilders = new ConditionalWeakTable<Assembly, ModuleBuilder>();
```

Comment:
I wonder if the assembly can actually be unloaded when dynamic types are referencing it. We should give it a try at some point

Reply:
we can make it work, but I'll leave that for another PR and another release, changed to `Dictionary` in 8bb619b